jax/numpy/lax_numpy.py

```diff
@@ -4222,36 +4221,34 @@ def _quantile(a, q, axis, interpolation, keepdims, squash_nans):
 
 
 @partial(jit, static_argnums=2)
-@partial(vectorize, excluded={0, 2})
-def _searchsorted(a, v, side):
-  op = operator.le if side == 'left' else operator.lt
-
-  def cond_fun(state):
-    start, stop = state
-    return stop - start > 1
-
-  def body_fun(state):
-    start, stop = state
-    mid = (start + stop) // 2
-    go_left = op(v, a[mid])
-    return where(go_left, start, mid), where(go_left, mid, stop)
-
-  result = lax.while_loop(cond_fun, body_fun, (0, a.shape[0]))
-  return where(op(v, a[0]), 0, result[1])
-
+def _searchsorted(a, v, side='left'):
+  op = operator.lt if side == "right" else operator.le
+
+  def fscan(carry, x):
+    low, high = carry
+    mid = (low + high) // 2
+    mask = op(v, a[mid])
+    low = where(mask, low, mid)
+    high = where(mask, mid, high)
+    return (low, high), x
```

Comment:
I think you could write `def fscan(cary, _)` and also just return `None` here, instead of passing through the value. It wouldn't make a difference except perhaps readability, though.

Reply:
Done.